### ISSUE TITLE
[FR] `SaveWhen` should be a `[Flags]` enum

### ISSUE THREAD

**Issue by Lordfirespeed** [MaxWasUnavailable/LethalModDataLib#2]
```cs
[Flags]
public enum SaveWhen 
{
    Manual = SaveType.None,
    OnSave = SaveType.Save,
    OnAutoSave = SaveType.Save | SaveType.AutoSave,
}
```

Which goes hand-in-hand with the proposed `SaveType` (see #1)
```cs
[Flags]
public enum SaveType
{
    None = 0,
    Save = 1 << 0,
    AutoSave = 1 << 1,
}
```

**Comment by MaxWasUnavailable** [comment]
Will add, should be simple enough to add to the next release.

**Comment by MaxWasUnavailable** [comment]
Closed by 9691a797 & bd808439 & d4b0c42557a385bee51ed4f540a45303dc41babd
